logstash-core/src/main/java/org/logstash/secret/store/SecretStoreUtil.java

```diff
@@ -96,4 +119,45 @@ private SecretStoreUtil() {
     public static void clearBytes(byte[] bytes) {
         Arrays.fill(bytes, (byte) '\0');
     }
+
+
+    /**
+     * De-obfuscates the obfuscated char[] generated by {@link SecretStoreUtil#obfuscate(char[])}
+     *
+     * @param chars The chars to de-obscure
+     * @return the de-obscured chars
+     */
+    public static char[] deObfuscate(char[] chars) {
+        byte[] bytes = asciiCharToBytes(chars);
+        byte[] random = Arrays.copyOfRange(bytes, bytes.length / 2, bytes.length);
```

Comment:
The first second half of the `chars` is  some kind of one-time pad? Can you add comments describing the format or pointing to a reference that describes the format?

Reply:
I struggled with how to document...since this is the obfuscation code and open source. Kinda trying to the make the code obtuse.

> The first second half of the chars is some kind of one-time pad? 
They first and second half are original bits (b1) and the XOR bits (b2) respectively.  b1 = XOR(b2) && b2 = XOR(b1).